### PR TITLE
Support temperature sensor SHT31

### PR DESCRIPTION
**Pull request by miaoski**
Hi.
This PR adds support to Adafruit SHT31. I have compiled it on the Docker environment and tested it on my Lilygo TBeam.
Thank you.

### CODE REVIEW

**Comment by miaoski** [comment]
Hi. I found that I should have changed Meshtastic Protobuf Definitions instead of the generated pb2, but it's in another repo. Allow me to be lazy here, if you may ...

```diff
diff --git a/telemetry.proto b/telemetry.proto
index 4234a8f..3d210cf 100644
--- a/telemetry.proto
+++ b/telemetry.proto
@@ -157,4 +157,9 @@ enum TelemetrySensorType {
   */
   QMC5883L = 11;
 
+  /*
+  * High accuracy temperature and humidity
+  */
+  SHT31 = 12;
+  
 };
```

**Comment by garthvh** [comment]
Needs to be in the protobufs repo or it will be overwritten the next time there is a protobufs change

**Comment by mc-hamster** [comment]
Echoing what Garth said. This needs to be in the protobufs and compiled.

Sorry for the extra steps. It's how we maintain the protocol definition across different platforms.

**Comment by miaoski** [comment]
Hi. It makes sense. Let me resubmit a PR with the patch in protobufs.

File: release/latest/.gitignore [hunk 1]
```diff
@@ -1 +0,0 @@
-curfirmwareversion.xml
```
Comment (caveman99) [inline release/latest/.gitignore:1]: please restore this file, it has nothing to do with the commit.

File: src/mesh/generated/telemetry.pb.h [hunk 2]
```diff
@@ -87,7 +89,7 @@ typedef struct _Telemetry {
 
 /* Helper constants for enums */
 #define _TelemetrySensorType_MIN TelemetrySensorType_SENSOR_UNSET
-#define _TelemetrySensorType_MAX TelemetrySensorType_QMC5883L
+#define _TelemetrySensorType_MAX TelemetrySensorType_SHT31
 #define _TelemetrySensorType_ARRAYSIZE ((TelemetrySensorType)(TelemetrySensorType_QMC5883L+1))
```
Comment (caveman99) [inline src/mesh/generated/telemetry.pb.h:93]: Please revert this change, as this file is auto-generated

**Comment by caveman99** [comment]
PR change is approved, please review my remarks.

**Comment by miaoski** [comment]
Hi @caveman99 ,
Thank you for the comment. Should I close this PR and submit a new PR that fixes the unnecessary changes?
